src/main/java/com/google/sps/servlets/CompleteTaskServlet.java

```diff
@@ -0,0 +1,91 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.gson.Gson;
+import com.google.sps.data.Task;
+import com.google.sps.data.User;
+import com.google.sps.data.Response;
+
+import java.util.HashMap;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing contacts. */
+@WebServlet("/complete-task")
+public class CompleteTaskServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    String userN = request.getParameter("userN");
+    String userID = "";
+    String currentTaskID = "";
+
+    // in the case where there is no userN parameter return an error 
+    if (userN == null || userN == "") {
+        response.sendError(400, "query parameter userN is not defined");
+        return;
+    }
+
+    Query<Entity> userQuery =
+    Query.newEntityQueryBuilder().setKind("User").build();
+    QueryResults<Entity> userResults = datastore.run(userQuery);
```

Comment:
this runs in O(number of users), we could change this to O(1) by using userN as the datastore key then doing a datastore get